vividus-to-xray-exporter/src/main/java/org/vividus/xray/exporter/XrayExporter.java

```diff
@@ -0,0 +1,256 @@
+public class XrayExporter
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(XrayExporter.class);
+
+    private static final String SEMICOLON = ";";
+
+    @Autowired private XrayExporterOptions xrayExporterOptions;
+    @Autowired private XrayFacade xrayFacade;
+    @Autowired private JiraFacade jiraFacade;
+
+    private final List<ErrorExportEntry> errors = new ArrayList<>();
+
+    public void exportResults() throws IOException
+    {
+        for (Story story : readStories())
+        {
+            LOGGER.atInfo().addArgument(story::getPath)
+                           .log("Exporting scenarios from {} story");
+
+            for (Scenario scenario : story.getScenarios())
+            {
+                String scenarioTitle = scenario.getTitle();
+
+                List<Meta> scenarioMeta = scenario.getMeta();
+                if (isSkipped(scenarioMeta))
+                {
+                    LOGGER.atInfo().addArgument(scenarioTitle)
+                                   .log("Skip export of {} scenario");
+                    continue;
+                }
+                LOGGER.atInfo().addArgument(scenarioTitle)
+                               .log("Exporting {} scenario");
+
+                String storyTitle = story.getPath();
+
+                TestCaseParameters parameters = new TestCaseParameters();
+                parameters.setLabels(getMetaValues(scenarioMeta, "xray.labels"));
+                parameters.setComponents(getMetaValues(scenarioMeta, "xray.components"));
+                parameters.setSummary(scenarioTitle);
+                parameters.setStory(storyTitle);
+                parameters.setSteps(scenario.collectSteps());
+
+                try
+                {
+                    String testCaseId = ensureOneValueOrNull(scenarioMeta, "testCaseId");
+                    if (testCaseId != null)
+                    {
+                        updateTestCase(testCaseId, parameters);
+                    }
+                    else
+                    {
+                        testCaseId = xrayFacade.createTestCase(parameters);
+                    }
+                    createTestsLink(testCaseId, scenarioMeta);
+                }
+                catch (IOException | SyntaxException | InvalidIssueStatusException e)
+                {
+                    errors.add(new ErrorExportEntry(storyTitle, scenarioTitle, e.getMessage()));
+                    LOGGER.atError().setCause(e)
+                                    .log("Got an error while exporting");
+                }
+            }
+        }
+
+        publishErrors();
+    }
+
+    public void publishErrors() throws IOException
+    {
+        if (!errors.isEmpty())
+        {
+            String message = new ObjectMapper().setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
+                                               .writerWithDefaultPrettyPrinter()
+                                               .writeValueAsString(errors);
```

Comment:
I believe the errors should be printed in user-friendly format (not technical JSON)